core/blockchain.go

```diff
@@ -3089,19 +3147,138 @@ func (bc *BlockChain) SubscribeBlockProcessingEvent(ch chan<- bool) event.Subscr
 }
 
 // Options
-func EnableLightProcessor(bc *BlockChain) *BlockChain {
+func EnableLightProcessor(bc *BlockChain) (*BlockChain, error) {
 	bc.processor = NewLightStateProcessor(bc.Config(), bc, bc.engine)
-	return bc
+	return bc, nil
 }
 
-func EnablePipelineCommit(bc *BlockChain) *BlockChain {
+func EnablePipelineCommit(bc *BlockChain) (*BlockChain, error) {
 	bc.pipeCommit = true
-	return bc
+	return bc, nil
 }
 
 func EnablePersistDiff(limit uint64) BlockChainOption {
-	return func(chain *BlockChain) *BlockChain {
+	return func(chain *BlockChain) (*BlockChain, error) {
 		chain.diffLayerFreezerBlockLimit = limit
-		return chain
+		return chain, nil
+	}
+}
+
+func EnableBlockValidator(chainConfig *params.ChainConfig, engine consensus.Engine, mode VerifyMode, peers verifyPeers) BlockChainOption {
+	return func(bc *BlockChain) (*BlockChain, error) {
+		if mode.NeedRemoteVerify() {
+			vm, err := NewVerifyManager(bc, peers, mode == InsecureVerify)
+			if err != nil {
+				return nil, err
+			}
+			go vm.mainLoop()
+			bc.validator = NewBlockValidator(chainConfig, bc, engine, EnableRemoteVerifyManager(vm))
+		}
+		return bc, nil
+	}
+}
+
+func (bc *BlockChain) GetRootByDiffHash(blockNumber uint64, blockHash common.Hash, diffHash common.Hash) *VerifyResult {
```

Comment:
The function name `GetRootByDiffHash` does not match with its return value: `*VerifyResult` .
Please try better name.

Reply:
> The function name `GetRootByDiffHash` does not match with its return value: `*VerifyResult` . Please try better name.

fixed

Reply:
`func (s *PublicBlockChainAPI) GetRootByDiffHash` is calling this function, its name need to be changed as well.

Reply:
> `func (s *PublicBlockChainAPI) GetRootByDiffHash` is calling this function, its name need to be changed as well.

fixed